chore(vercel): デプロイフロー整備＋ナビゲーション修正（CTA/Header/Hero）＆ Prisma 対応

Kazuya-Sakashita opened the pull request:
# chore(vercel): デプロイフロー整備＋ナビゲーション修正（CTA/Header/Hero）＆ Prisma 対応

## 🧾 概要
- Vercel デプロイを安定化するために **Deploy Hook を叩く GitHub Actions** を追加
- Vercel の依存キャッシュ対策として **Prisma Client を自動生成**（`postinstall: prisma generate`）
- **`@prisma/client` と `prisma` のバージョン整合**（6.15.0 に固定）
- ランディングの **CTA / Header / Hero のボタンを正規のリンク遷移に修正**
- `useNavigation` の **重複キー削除**（`no-dupe-keys` 解消）

---

## ✨ 変更点（Diff サマリ）
### 1) CI / Deploy
- `.github/workflows/vercel-deploy-hook.yml` を新規追加  
  - `develop` への push をトリガに **Vercel Deploy Hook** を起動  
  - `secrets.VERCEL_DEPLOY_HOOK_URL` を使用

### 2) Prisma / package.json
- `scripts.postinstall` に `prisma generate` を追加  
- `@prisma/client` を `6.15.0` に、`prisma` を `6.15.0` に固定（整合）  
- それ以外のスクリプトは維持

### 3) ランディング UI 遷移の正規化
- `src/app/_components/home/CTA.tsx`
  - Button を **`<Link>` でラップ**（`/dashboard`・`/navigator` へ遷移）
- `src/app/_components/home/Header.tsx`
  - `ManaboIcon` を使用
  - 「始める」ボタンを **`<Button asChild><Link .../></Button>`** に変更（`/dashboard`）
- `src/app/_components/home/Hero.tsx`
  - 「無料で始める」ボタンを **`<Button asChild><Link .../></Button>`** に変更（`/dashboard`）
  - `ArrowRight` に `aria-hidden` を付与

### 4) useNavigation の重複削除
- `src/app/_hooks/useNavigation.ts`
  - `type` と `return` オブジェクトの **`onNavigateToDashboard` 重複定義を削除**  
  - ESLint `no-dupe-keys` / TS 1117 を解消

---

## 🔧 技術的ポイント
- **Prisma × Vercel** で発生する「クライアント未生成」問題に対し、`postinstall` で毎回生成させることで解消  
- リンク遷移は **Next.js の `<Link>` を正として、ボタンは `asChild` でアクセシブルに**  
- 今後の ESLint 警告（`no-console` / import 周り）は別 PR で段階的に解消予定

---

## ✅ 動作確認
- [x] ローカルで `npm ci && npm run build` が成功する  
- [x] `npm run lint` でエラーが出ない（警告は現状維持）  
- [x] `/`（ランディング）から **「無料で始める」→ `/dashboard`** へ遷移できる  
- [x] Header の **「始める」→ `/dashboard`** へ遷移できる  
- [x] CTA の **「機能をもっと見る」→ `/navigator`** へ遷移できる  
- [x] `develop` へ push 時、Actions が Vercel Deploy Hook を叩く

---

## 🧩 影響範囲
- ランディング周辺 UI（CTA / Header / Hero）の遷移先  
- デプロイフロー（GitHub Actions 追加・Vercel 側 Hook URL 依存）  
- Prisma Client 生成タイミング（`postinstall`）

---

## 🔐 環境変数・Secrets
- GitHub Secrets: `VERCEL_DEPLOY_HOOK_URL` を設定済みであること

---

## 🔗 関連
- Issue: #90（Vercel ビルドエラー対応）  
  ※番号は実プロジェクトに合わせて修正ください

---

## 📝 メモ
- Prisma のバージョンは **`@prisma/client` と `prisma` を常に同一**に保つ方針  
- 今回は `6.15.0` に統一（必要に応じて上げる際も **同時に** 変更）

Kazuya-Sakashita committed:
feat(home): CTAボタンを<Link>で実装し、/dashboard と /navigator へ遷移

Tomoe-Yoko committed:
ヘッダーアイコンを変更

Kazuya-Sakashita committed:
「無料で始める」、「始める」から /dashboard へ遷移

Kazuya-Sakashita committed:
fix(useNavigation): onNavigateToDashboard の重複定義を削除しビルドエラーを解消

- return オブジェクト内の重複キーを削除
- ESLint (no-dupe-keys) / TypeScript エラーを解消
- ビルドおよび lint の通過を確認

Kazuya-Sakashita committed:
chore(prisma): Vercel デプロイ時に Prisma Client を自動生成するよう修正

- package.json に postinstall スクリプトを追加し、`prisma generate` を自動実行
- @prisma/client と prisma のバージョンを同一に揃えて不整合を解消
- Vercel 環境で Prisma Client 未生成エラーが発生しないよう改善

Kazuya-Sakashita committed:
ci: Vercel デプロイフック用 GitHub Actions を追加

- .github/workflows/vercel-deploy-hook.yml を新規作成
- develop ブランチ更新時に Vercel Deploy Hook を叩いて自動デプロイを実行
- Secrets に登録した VERCEL_DEPLOY_HOOK_URL を利用

s-yamauchi07 commented on src/app/_components/home/CTA.tsx:
```diff
@@ -12,18 +11,22 @@ export const CTA = () => (
-      >
-        機能をもっと見る
-      </Button>
+      <Link href='/navigator'>
+        <Button
+          variant='outline'
+          className='w-full border-green-500 text-green-600 hover:bg-green-50 py-3 bg-transparent'
+          style={{ borderImage: 'linear-gradient(to right, #10b981, #059669) 1' }}
+        >
+          機能をもっと見る
+        </Button>
+      </Link>
```
```
      <Link href='/navigator'>
        <Button
          variant='outline'
          className='w-full border-green-500 text-green-600 hover:bg-green-50 py-3 bg-transparent'
          style={{ borderImage: 'linear-gradient(to right, #10b981, #059669) 1' }}
        >
          機能をもっと見る
        </Button>
      </Link>
 ```
遷移のパスは問題ないのですが、「機能をもっとみる」のボタンと`<BottomNavigation /> `が重なってボタンがうまく押せないので、認証前(ログイン前)のユーザーの場合は`<BottomNavigation /> `が見えないように実装いただくと良いかなと思いました。
こちらの[Issue89](https://github.com/shiftb-hub/manabo/issues/89)で修正いただく想定であれば、こちらのブランチではそのままでOKです！

Kazuya-Sakashita replied:
[fix(layout): ボトムバーとページ下部ボタンの重なりを解消 (#XX)](https://github.com/shiftb-hub/manabo/pull/91/commits/6e41be12eb33519077f90ed3061e2298a1c45597)

こちらで解消しました。確認ください。
<img width="299" height="628" alt="スクリーンショット 2025-09-19 7 00 44" src="https://github.com/user-attachments/assets/919562b6-4948-4023-8988-9945603f781e" />
<img width="1127" height="722" alt="スクリーンショット 2025-09-19 7 00 59" src="https://github.com/user-attachments/assets/d7a0931e-6eae-4ed5-b4f6-24dc2bde2a1f" />

s-yamauchi07 reviewed (commented):
1点確認コメント記載しました。
内容のご確認お願いいたします！

Kazuya-Sakashita committed:
fix(layout): ボトムバーとページ下部ボタンの重なりを解消 (#XX)

- RootLayout にボトムバー高さ分の下余白を追加
- CSS変数 (--bottom-nav-h) を導入して高さを管理
- iOS セーフエリア (env(safe-area-inset-bottom)) に対応

s-yamauchi07 reviewed (approved):
LGTMです！
